pkg/build/builder/util.go

```diff
@@ -150,3 +153,69 @@ func ParseProxyURL(proxy string) (*url.URL, error) {
 
 	return proxyURL, err
 }
+
+// CopyDirIfExists recursively copies a directory to the destination path.
+// If the source directory does not exist, no error is returned.
+// If the destination directory exists, any contents with matching file names
+// will be overwritten.
+func CopyDirIfExists(src, dst string) error {
+	srcInfo, err := os.Stat(src)
+	if os.IsNotExist(err) {
+		return nil
+	}
+	if err = os.MkdirAll(dst, srcInfo.Mode()); err != nil {
+		return err
+	}
+	dirInfo, err := ioutil.ReadDir(src)
+	for _, info := range dirInfo {
+		srcPath := filepath.Join(src, info.Name())
+		dstPath := filepath.Join(dst, info.Name())
+		switch info.Mode() & os.ModeType {
+		case os.ModeDir:
+			err = CopyDirIfExists(srcPath, dstPath)
+		case os.ModeSymlink:
+			err = CopySymLink(srcPath, dstPath)
+		default:
+			err = CopyFileIfExists(srcPath, dstPath)
+		}
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// CopyFileIfExists copies the source file to the given destination, if the source file exists.
+// If the destination file exists, it will be overwritten and will not copy file attributes.
+func CopyFileIfExists(src, dst string) error {
+	_, err := os.Stat(src)
+	if os.IsNotExist(err) {
+		return nil
+	}
+	in, err := os.Open(src)
+	if err != nil {
+		return err
+	}
+	defer in.Close()
+
+	out, err := os.Create(dst)
+	if err != nil {
+		return err
+	}
+	defer out.Close()
+
+	_, err = io.Copy(out, in)
+	if err != nil {
+		return err
+	}
+	return out.Close()
+}
+
+// CopySymLink preserves symlinks as the form of copy
+func CopySymLink(src, dst string) error {
+	link, err := os.Readlink(src)
```

Comment:
What if the symlink refers to an absolute path, instead of a relative path? Do we keep as is?

See also https://github.com/openshift/source-to-image/blob/2ed02f2351b48858b5cc358bbccf6fc4a2a284ef/pkg/util/fs/fs.go#L208-L229